src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStep.java

```diff
@@ -36,40 +37,65 @@
 
     private static final String DEFAULT_CLOUD = "kubernetes";
 
+    @CheckForNull
     private String cloud = DEFAULT_CLOUD;
+
+    @CheckForNull
     private String inheritFrom;
 
+    @CheckForNull
     private String label;
+
+    @CheckForNull
     private String name;
 
+    @CheckForNull
     private String namespace;
+
     private List<ContainerTemplate> containers = new ArrayList<>();
     private List<TemplateEnvVar> envVars = new ArrayList<>();
     private List<PodVolume> volumes = new ArrayList<PodVolume>();
+
+    @CheckForNull
     private WorkspaceVolume workspaceVolume;
+
     private List<PodAnnotation> annotations = new ArrayList<>();
     private List<String> imagePullSecrets = new ArrayList<>();
 
-    private int instanceCap = Integer.MAX_VALUE;
+    private int instanceCap;
```

Comment:
I think this is causing

```
o.c.j.p.k.KubernetesCloud#addProvisionedSlave: Maximum number of concurrently running agent pods (0) reached for template docker_1-b72k5-81dhh in Kubernetes Cloud kubernetes, not provisioning: 0 running or pending in namespace kubernetes-plugin-test with label "docker_1-b72k5" and Kubernetes labels {jenkins=slave, BUILD_NUMBER=undefined, jenkins/label=docker_1-b72k5, test=docker, BRANCH_NAME=undefined, class=ContainerExecDecoratorPipelineTest}
```

Reply:
https://github.com/jenkinsci/kubernetes-plugin/blob/f7e84d5370711ace15c74118515a2e2a0ddf8d8a/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java#L586-L592 should have a quick exit when `instanceCap == 0`